src/osd/OSD.cc

```diff
@@ -9768,9 +9768,33 @@ int OSD::init_op_flags(OpRequestRef& op)
 }
 
 void OSD::set_perf_queries(const std::list<OSDPerfMetricQuery> &queries) {
+  dout(10) << "setting " << queries.size() << " queries" << dendl;
+
+  m_perf_queries = queries;
```

Comment:
Nit: perhaps ```OSD::handle_pg_create_info``` needs to be updated as well to set the queries when a new PG is created (due to a split, pool creation, etc).